src/vnode.ts

```diff
@@ -12,22 +14,32 @@ export interface VNode {
 }
 
 export interface VNodeData {
-  // modules - use any because Object type is useless
-  props?: any;
-  attrs?: any;
-  class?: any;
-  style?: any;
-  dataset?: any;
-  on?: any;
-  hero?: any;
-  attachData?: any;
+  props?: {
+    [prop: string]: any
+  };
+  attrs?: {
+    [attr: string]: any
```

Comment:
I think we can restrict this one to `string | number | boolean`. Attributes are always strings, numbers coerce to strings, and booleans are for boolean attributes.